File: csharp/knapsack/Knapsack.cs
public static class Knapsack
{
    public static int MaximumValue(int maximumWeight, (int weight, int value)[] items)
    {
        int numItems = items.Length;
        int[,] dp = new int[numItems + 1, maximumWeight + 1];
        for (int i = 1; i <= numItems; i++)
        {
            var currentItem = items[i - 1];
            for (int w = 1; w <= maximumWeight; w++)
            {
                if (currentItem.weight > w)
                {
                    dp[i, w] = dp[i - 1, w];
                }
                else
                {
                    dp[i, w] = Math.Max(dp[i - 1, w], currentItem.value + dp[i - 1, w - currentItem.weight]);
                }
            }
        }
        return dp[numItems, maximumWeight];
    }
}

File: javascript/knapsack/knapsack.js
//
// This is only a SKELETON file for the 'Knapsack' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const knapsack = (maximumWeight, items) => {
  const dp = Array(maximumWeight + 1).fill(0);

  for (const { weight, value } of items) {
    for (let w = maximumWeight; w >= weight; w--) {
      dp[w] = Math.max(dp[w], dp[w - weight] + value);
    }
  }
  return dp[maximumWeight];
};
